apps/microtvm/zephyr/template_project/microtvm_api_server.py

```diff
@@ -0,0 +1,692 @@
+    for d in (API_SERVER_DIR / "src").iterdir():
+        if d.is_dir():
+            PROJECT_TYPES.append(d.name)
+
+
+PROJECT_OPTIONS = [
+    server.ProjectOption(
+        "extra_files",
+        help="If given, during generate_project, uncompress the tarball at this path into the project dir",
+    ),
+    server.ProjectOption(
+        "gdbserver_port", help=("If given, port number to use when running the local gdbserver")
+    ),
+    server.ProjectOption(
+        "nrfjprog_snr",
+        help=(
+            "When used with nRF targets, serial # of the " "attached board to use, from nrfjprog"
+        ),
+    ),
+    server.ProjectOption(
+        "openocd_serial",
+        help=("When used with OpenOCD targets, serial # of the " "attached board to use"),
+    ),
+    server.ProjectOption(
+        "project_type",
+        help="Type of project to generate.",
+        choices=tuple(PROJECT_TYPES),
+    ),
+    server.ProjectOption("verbose", help="Run build with verbose output"),
+    server.ProjectOption(
+        "west_cmd",
+        help=(
+            "Path to the west tool. If given, supersedes both the zephyr_base "
+            "option and ZEPHYR_BASE environment variable."
+        ),
+    ),
+    server.ProjectOption("zephyr_base", help="Path to the zephyr base directory."),
+    server.ProjectOption("zephyr_board", help="Name of the Zephyr board to build for"),
+]
+
+
+class Handler(server.ProjectAPIHandler):
+    def __init__(self):
+        super(Handler, self).__init__()
+        self._proc = None
+
+    def server_info_query(self, tvm_version):
+        return server.ServerInfo(
+            platform_name="zephyr",
+            is_template=IS_TEMPLATE,
+            model_library_format_path=""
+            if IS_TEMPLATE
+            else (API_SERVER_DIR / MODEL_LIBRARY_FORMAT_RELPATH),
+            project_options=PROJECT_OPTIONS,
+        )
+
+    # These files and directories will be recursively copied into generated projects from the CRT.
+    CRT_COPY_ITEMS = ("include", "Makefile", "src")
+
+    def _create_prj_conf(self, project_dir, options):
+        with open(project_dir / "prj.conf", "w") as f:
+            f.write(
+                "# For UART used from main().\n"
+                "CONFIG_RING_BUFFER=y\n"
+                "CONFIG_UART_CONSOLE=n\n"
+                "CONFIG_UART_INTERRUPT_DRIVEN=y\n"
+                "\n"
+            )
+            f.write("# For TVMPlatformAbort().\n" "CONFIG_REBOOT=y\n" "\n")
+
+            if True:  # options["project_type"] == "host_driven":
+                f.write("# For RPC server C++ bindings.\n" "CONFIG_CPLUSPLUS=y\n" "\n")
+
+            f.write("# For math routines\n" "CONFIG_NEWLIB_LIBC=y\n" "\n")
+
+            if self._has_fpu(options["zephyr_board"]):
+                f.write("# For models with floating point.\n" "CONFIG_FPU=y\n" "\n")
+
+            main_stack_size = None
+            if self._is_qemu(options) and options["project_type"] == "host_driven":
```

Comment:
`aot_demo` condition is missing?
I think for that the stack size should be 2000

Reply:
@mehrdadh I think CI won't catch the `tutorial/micro/micro_tflite.py`. On the other hand I'm ok fixing it on a follow-up PR.

Reply:
@mehrdadh the API server does work on physical boards, I've tested it on disco boards, for instance. `test_zephyr` and `test_zephyr_aot` defaults to use qemu as on the old API, but `prj.conf` generator will take care of generating the proper config file for physical boards too.

Reply:
@mehrdadh actually, I've tested the previous version, so I'm re-testing it on disco. Hence it's good to test as you said with the 4 additional boards you have on hand.